README.md

```diff
@@ -1,77 +1,92 @@
-  How to do code review:
-  As a reviewer- Todo: link
-  As a reviewee-Todo: link
-  Code reviews on GitHub- Todo: link
-  
-  ### Finishing homework
-  You can consider your homework complete if all of the above steps pass. 
-  **Don't merge the PR to master branch**
-  A single best homework implementation will be merged to master. 
-
-  
-
-  # Community
-  If you want to join as either mentor or pupil or have any questions, please refer to our discord Community.  
-  https://discord.gg/rCMKcUU
-  
-  
-
-
-
- 
-
- 
 
+# Homework 4
+
+## Hints
+* Do use Environment.NewLine if you want to concanate a string with a newline command (or use WriteLine).
+* Be very careful where to place your NewLine's and where not!
+* Do use [CultureInfo.InvariantCulture](https://docs.microsoft.com/de-de/dotnet/api/system.globalization.cultureinfo.invariantculture?view=netcore-3.1) when reading in float numbers. (Due to differences in parsing numbers with '.' and ',' character.
+* The better your design, the easier it will be to find and fix errors!
+* Be careful when it comes to the BMI. You have to use cm and m for the height.
+* Do not hesitate to ask mentors for advice. This homework IS puzzling.
+
+## Task
+Modify your old functions to fit the specification:
+1) Validate number input:
++ Input is not a number -> return -1.
++ Input is a number but 0 -> return 0.
++ Input is valid -> return input.
+2) Validate name input:
++ Empty string for name -> return "-".
++ Print error message "Name cannot be empty." in a new line.
+3) Validate BMI input:
++ Height and/or weight being less or equal 0 -> return -1.
++ Print error messages:
+  - "Failed calculating BMI. Reason:"
+  - "Height cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
+  - "Weight cannot be equal or less than zero, but was X.", where X is the console input (X <= 0)
++ If height and weight are invalid print both messages, one in a new line each.
+  - The error message of height does **NOT!!!!** contain **"equal or "** then!
+
+## Tests Summary
+The test results should be as pointed out below.
+
+A stands for a valid value.
+X, Y stand for invalid values.
+R stands for valid value but not in the wanted range.
+E stand for empty input string. (Just pressing enter when facing prompt.)
+M stands for string message.
+
+Format:
+* Input -> return value
+  * Output on same line as input.
+* ~> User/Test input being A if valid, X if invalid.
+* Output2 on new line.
+  * Output3 on same line as Output2.
+
+Checks.CalculateBmi(A, X) -> -1
+* "Failed calculating BMI. Reason:"
+* "Height cannot be equal or less than zero, but was X."
+* ""
+
+Checks.CalculateBmi(X, A) -> -1
+* "Failed calculating BMI. Reason:"
+* "Weight cannot be equal or less than zero, but was X."
+* ""
+
+Checks.CalculateBmi(X, Y) -> -1
+* "Failed calculating BMI. Reason:"
+* "Height cannot be less than zero, but was X."
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4885761)